Brand colors

jimmynotjim opened the pull request:
Updated all the brand colors so we can update this project to CFv4.

## Changes

- Updated the color palette to match latest
- Updated the color swatches to utilize a template that pulls the colors from the CSV file
- Updated the source code link

## Additions

- Added CSV file that is exported from the color spec document
- Added a Less variables generation script that pulls the colors from the CSV file

## Testing

- Run `./setup.sh` to install new dependencies and rebuild
- Run `npm run colors` to regenerate the color palette
- Run `grunt less` and `npm start`
- Navigate to http://localhost:4000/brand-guidelines/color-principles.html 

## Review

- @Scotchester 
- @designlanguage 
- @nataliafitzgerald 

[Preview this PR without the whitespace changes](?w=0)

## Notes

## Todos

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

Scotchester commented on brand-guidelines/color-principles.md:
```diff
@@ -24,7 +24,7 @@ A minimal palette, clear hierarchy, good information design, and ample whitespac
 </div>
 
-<h2 id="palette">Palette<span class="cf-code-link"><a href="https://github.com/cfpb/cf-colors/blob/gh-pages/src/cf-colors.less">View code <i class="cf-icon cf-icon-external-link"></i></a></span></h2>
+<h2 id="palette">Palette<span class="cf-code-link"><a href="https://github.com/cfpb/generator-cf/blob/master/app/templates/src/static/css/brand-palette.less">View code <i class="cf-icon cf-icon-external-link"></i></a></span></h2>
```
Please update this link to https://github.com/cfpb/cf-theme-cfpb/blob/master/src/color-palette.less. Even if cf-theme-cfpb doesn't take hold, I really don't want the generator to be considered the canonical location of the palette.

Scotchester reviewed (changes requested):
Before merging this, I want to be sure that (at least some) of our graphic designers are okay with having the print colors be blank until the press checks are complete and we've locked down the formulations.

@designlanguage @nataliafitzgerald @caheberer Are you all okay with that?

Scotchester commented:
> It didn't make sense to add the other tints of green to the primary palette. Not sure if we should re-work the primary, background, secondary lists or what, because grays aren't always backgrounds and I'd argue `@green-20` isn't a primary brand color.

I agree that we should re-think how the palette is framed on the color principles page.

jimmynotjim commented:
From @designlanguage 

> Rather than “CMYK coming soon”, can we just not include CMYK until we have those values?
>
> I agree with Jimmy’s comments about the color groupings. We can show just our green as the primary color, but I don't think the tints need to be excluded. 
> 
> Another factor is that with the added tints for all colors, it wouldn't neatly fit in the four columns of the current design. How about a different tabular format... something like this:

![screen shot 2017-03-02 at 5 48 22 pm](https://cloud.githubusercontent.com/assets/1280430/23569723/75770a36-0026-11e7-9594-e7d6bd0b1e3c.png)

> Just to clarify the groupings... I think we only need primary and secondary colors (along with respective tints).

jimmynotjim commented:
@Scotchester @nataliafitzgerald @designlanguage If I take care of the CMYK can we merge it as is and backlog the layout? That's really a task all to itself.

jimmynotjim committed:
Updated the color palette to match latest

- Added CSV file that is exported from the color spec document
- Added a Less variables generation script that pulls the colors from the CSV file
- Updated the color swatches to utilize a template that pulls the colors from the CSV file
- Updated the source code link

jimmynotjim commented:
![screen shot 2017-03-07 at 10 58 42 am](https://cloud.githubusercontent.com/assets/1280430/23667916/0d26296a-0325-11e7-8fa1-ba646c4ce0e1.png)

Scotchester commented:
I thought you wanted to merge this before doing all that work? 😜

jimmynotjim commented:
@nataliafitzgerald @designlanguage here's a preview of what the new layout looks like (ignore the misalignments between the three screenshots)

![screen shot 2017-03-16 at 11 40 56 am](https://cloud.githubusercontent.com/assets/1280430/24007734/801d542e-0a3d-11e7-9686-2726abadef25.png)
![screen shot 2017-03-16 at 11 41 06 am](https://cloud.githubusercontent.com/assets/1280430/24007739/825d6238-0a3d-11e7-8a79-d324d6c13c30.png)
![screen shot 2017-03-16 at 11 41 15 am](https://cloud.githubusercontent.com/assets/1280430/24007741/844b1068-0a3d-11e7-95cf-6dec8fedb104.png)

Scotchester commented:
That is looking :fire:

Scotchester commented:
You might already be planning this, but how about we format the hex value in a fixed-with font?

designlanguage commented:
I don't think Green 60 and Green 20 should be primary colors. It could just be CFPB green and Black, (possibly with each spanning 2 columns, with a swatch as thin as the secondary colors).

designlanguage commented:
Besides that, I agree that this is looking great @jimmynotjim and @Scotchester !

designlanguage commented:
One other thought: the body copy under primary color doesn't need to mention whitespace. It's already mentioned above in the text under Color (but maybe that's just placeholder text anyway).

jimmynotjim commented:
~~Updated the font-face of the values to monospace per @Scotchester~~

@designlanguage that's the exact copy from the GDoc. Let's update it there to what you want for both that and the tints and I'll copy it over.

jimmynotjim commented:
@designlanguage here's what the primary colors look like at 2 columns and the same height as the others

![screen shot 2017-03-16 at 1 24 08 pm](https://cloud.githubusercontent.com/assets/1280430/24012400/df8e28b2-0a4b-11e7-8fcf-fcb8f52764fd.png)

jimmynotjim commented:
Bah! @designlanguage realized I was looking at the wrong text that you had commented on. I've removed the repetitive text from under the "Primary" heading and removed the placeholder text from under the "Tint" heading.

jimmynotjim commented:
@Scotchester do you want to run this locally as a sanity check before we merge it?

Scotchester commented:
@jimmynotjim Yes I do. Ready for me to give it a go?

Scotchester commented on scripts/npm/colors/index.js:
```diff
@@ -0,0 +1,33 @@
+  if ( data[0] === 'family' ) {
+    return fs.readFileSync( templateSource )
+             .toString()
+             .replace( /\[date\]/g, moment().format( 'MMMM Do YYYY, h:mm:ss a' ) );
```
I think we're gonna want to change this. Any accidental/unnecessary run of this script will update the timestamp, even if no variables changed.

jimmynotjim replied:
There's no really way to accidentally run the script. You have to purposefully do it and we can catch that change in a PR. This is definitely the easiest way, otherwise you have to update the template manually and it's yet another step to remember. I'm trying to remove steps.

Scotchester replied:
I guess that's true, yeah.

Scotchester commented on src/static/css/content.less:
```diff
@@ -316,6 +322,19 @@ figure {
+
+    .swatch_field {
+        width: 200px;
+        height: 2.5em;
```
Not a blocker to release if we need to have a discussion about it, but I'd like to add `border-bottom: 1px solid @white;` here, to override the row's standard gray border between color swatches.

jimmynotjim replied:
I think we should keep the gray only because you can't tell the white is even there without the border above/below.

Scotchester replied:
I'm not sure I follow, or I'm not sure you follow. This is what I was envisioning:

![screen shot 2017-03-16 at 17 48 38](https://cloud.githubusercontent.com/assets/1044670/24020022/e45bc9a6-0a70-11e7-8f7a-4dbf23aabc73.png)

I just feels cleaner to me than the gray.

designlanguage replied:
I prefer the white to the gray line, but can there be no line (or a line that's the same color as the swatch above it)? 

Separately, it would be great if we could adjust the spacing of the columns a bit. The names are pretty tight against the swatch compared to the spacing between Name->HEX or GRB->CMYK.

Scotchester commented on brand-guidelines/color-principles.md:
```diff
@@ -24,945 +24,109 @@ A minimal palette, clear hierarchy, good information design, and ample whitespac
 
```
This is as high as I can place this comment, but please update the page intro text above here to reflect the latest in the working doc.

jimmynotjim replied:
Fixed.

Scotchester commented on brand-guidelines/color-principles.md:
```diff
@@ -24,945 +24,109 @@ A minimal palette, clear hierarchy, good information design, and ample whitespac
 
-<h2 id="palette">Palette<span class="cf-code-link"><a href="https://github.com/cfpb/cf-colors/blob/gh-pages/src/cf-colors.less">View code <i class="cf-icon cf-icon-external-link"></i></a></span></h2>
+<h2 id="palette">Palette<span class="cf-code-link"><a href="https://github.com/cfpb/cf-theme-cfpb/blob/master/src/color-palette.less">View code <i class="cf-icon cf-icon-external-link"></i></a></span></h2>
 
```
There should be a sentence here. Refer to the working doc.

jimmynotjim replied:
Fixed.

Scotchester commented on brand-guidelines/color-principles.md:
```diff
@@ -24,945 +24,109 @@ A minimal palette, clear hierarchy, good information design, and ample whitespac
+                    <table class="swatch_table">
+                        <tbody>
+                            <tr>
+                                <th>HEX</th>
```
Trivial, due to how it's styled, but this should be "Hex".

Scotchester commented on brand-guidelines/color-principles.md:
```diff
@@ -24,945 +24,109 @@ A minimal palette, clear hierarchy, good information design, and ample whitespac
+                    <table class="swatch_table">
+                        <tbody>
+                            <tr>
+                                <th>HEX</th>
```
Same comment about "Hex".

Scotchester commented on brand-guidelines/color-principles.md:
```diff
@@ -24,945 +24,109 @@ A minimal palette, clear hierarchy, good information design, and ample whitespac
-    </figure>
-</div> <!-- /.swatches -->
+    {% for color in site.data.cfpb-brand-colors %}
+        {% if color.primary == "true" %}
```
Trivial, but this could just be `{% if color.primary %}`

jimmynotjim replied:
My only worry is if someone added an entry there by accident, but I can change it.

Scotchester commented on brand-guidelines/color-principles.md:
```diff
@@ -24,945 +24,109 @@ A minimal palette, clear hierarchy, good information design, and ample whitespac
-        </figcaption>
-    </figure>
+    {% for color in site.data.cfpb-brand-colors %}
+        {% if color.secondary == "true" %}
```
Same comment about simplifying the conditional.

Scotchester commented on brand-guidelines/color-principles.md:
```diff
@@ -24,945 +24,109 @@ A minimal palette, clear hierarchy, good information design, and ample whitespac
-        </figcaption>
-    </figure>
-</div>
+    {% assign family = 'green' %}
```
This could be `{% assign family = site.data.cfpb-brand-colors[1].family %}` to avoid hardcoding this value so it doesn't break if we reorder the CSV.

jimmynotjim replied:
Fixed.

Scotchester commented on brand-guidelines/color-principles.md:
```diff
@@ -24,945 +24,109 @@ A minimal palette, clear hierarchy, good information design, and ample whitespac
+        </thead>
+    {% for color in site.data.cfpb-brand-colors %}
+        {% if family != color.family %}
+        {% assign family = color.family %}
```
Could you please adjust the indentation so the conditional is easier to follow?

jimmynotjim replied:
Fixed.

Scotchester commented on brand-guidelines/color-principles.md:
```diff
@@ -24,945 +24,109 @@ A minimal palette, clear hierarchy, good information design, and ample whitespac
+        <thead>
+            <th></th>
+            <th>Name</th>
+            <th>HEX</th>
```
Same comment about "Hex"

Scotchester commented on brand-guidelines/color-principles.md:
```diff
@@ -24,945 +24,109 @@ A minimal palette, clear hierarchy, good information design, and ample whitespac
+            <thead>
+                <th></th>
+                <th>Name</th>
+                <th>HEX</th>
```
Same comment about "Hex"

Scotchester commented on src/static/css/content.less:
```diff
@@ -279,6 +276,10 @@ figure {
 @media only screen and (min-width: 37.5em) {
+
+    th {
+        width: 70px;
```
Minor: Percentage value, perhaps?

jimmynotjim replied:
It looks better when it's consistent across devices sizes. That said, I've updated it to ems.

Scotchester commented on src/static/css/content.less:
```diff
@@ -288,20 +289,25 @@ figure {
+         'dark-gold', 'gold', 'gold-80', 'gold-60', 'gold-40', 'gold-20',
+         'dark-neutral', 'neutral', 'neutral-80', 'neutral-60', 'neutral-40', 'neutral-20',
+         'black', 'dark-gray', 'gray', 'gray-80', 'gray-60', 'gray-40', 'gray-20',
+         'gray-10', 'gray-5', 'white';
```
We could stop hard-coding this by having Jekyll process this file.

jimmynotjim replied:
Yeah, this is on my "to fix" list but in another PR. This is already a bit more complex than I'd like.

Scotchester commented on src/static/css/content.less:
```diff
@@ -316,6 +322,19 @@ figure {
 
+.color-table {
+    width: 100%;
+    margin-bottom: 40px;
```
Minor: Convert to ems and use a more standard value for us (i.e., a multiple of 15).

jimmynotjim replied:
Fixed.

Scotchester commented on src/static/css/content.less:
```diff
@@ -316,6 +322,19 @@ figure {
+    margin-bottom: 40px;
+
+    .swatch_field {
+        width: 200px;
```
Percentage width?

jimmynotjim replied:
Fixed.

Scotchester commented on src/static/css/content.less:
```diff
@@ -279,6 +276,10 @@ figure {
 
```
Need to work on the responsiveness for XS screens:

![screen shot 2017-03-16 at 17 31 25](https://cloud.githubusercontent.com/assets/1044670/24019526/b82639d6-0a6e-11e7-9c99-284f2c7befed.png)

jimmynotjim replied:
It's due to the tables. Once we get this on v4 I'd like to update the table to use the layout we're currently using on cfgov-refresh. That said, I think it's beyond the scope of this PR.

designlanguage replied:
@Scotchester - Does the tint-swatch table convert to a single column in mobile like the secondary colors?

Scotchester replied:
@designlanguage No, which is actually a more serious responsive issue than the one I screenshotted, but I agree that that can be remedied more easily with CF4 in place.

jimmynotjim replied:
The screenshot is caused by the tint table. If you delete it from the page, this section goes back to what it should be. The table itself forces the body wider than it should be.

Scotchester replied:
@jimmynotjim Ahhh, of course.

Scotchester reviewed (changes requested):
Found a fair bit of things, mostly minor or trivial.

jimmynotjim commented:
@Scotchester I think I've addressed everything.

Scotchester commented on brand-guidelines/color-principles.md:
```diff
@@ -98,35 +98,35 @@ These colors are used primarily as accent colors, and should be used to add vari
 
 {::nomarkdown}
-    {% assign family = 'green' %}
+    {% assign family = site.data.cfpb-brand-colors[1].family %
```
Lost a closing brace, here.

jimmynotjim replied:
Refresh. I fixed it here https://github.com/cfpb/design-manual/pull/457/commits/8de0b3e2f1fd9872dca9f339e17400c81dddb912

Scotchester replied:
Whoops, I was only looking at the first of your post-review commits.

Scotchester reviewed (changes requested):
One last tiny, but important, thing

Scotchester commented:
Great work on this! :shipit:
